sklearn/inspection/tests/test_partial_dependence.py

```diff
@@ -158,8 +158,10 @@ def test_grid_from_X_error(grid_resolution, percentiles, err_msg):
     (LinearRegression(), 'brute'),
     (GradientBoostingRegressor(random_state=0), 'brute'),
     (GradientBoostingRegressor(random_state=0), 'recursion'),
-    (HistGradientBoostingRegressor(random_state=0), 'brute'),
-    (HistGradientBoostingRegressor(random_state=0), 'recursion')]
+    (HistGradientBoostingRegressor(random_state=0, n_iter_no_change=None),
+     'brute'),
+    (HistGradientBoostingRegressor(random_state=0, n_iter_no_change=None),
```

Comment:
why would the test fail with early stopping?

Reply:
there are very small training set where early stopping leads to bad results

Reply:
the dataset has 100 samples (the default value for our `make_regression`), right? I kinda feel like the default parameters should give reasonable results for the default parameters of our data generators. Would that be too hard to have an `auto` early stopping parameter to figure if it's a good idea or not?

I really miss having a nice and more dynamic early stopping parameter, like a policy, a function, or an object.

Reply:
I think I remember a dataset with 40 samples. Not sure.

> Would that be too hard to have an auto early stopping parameter to figure if it's a good idea or not?

I don't think that's the place for this discussion.

See also the latest comments on #11324

Reply:
The tests fails with early stopping, otherwise I would not have changed this (I didn't know about the `inspection` module before). I can undo the changes locally to see where it fails exactly if you are interested.

Reply:
@NicolasHug I think this is a relevant place to discuss this here, because clearly we're changing our defaults in a way that makes the defaults break on a lot of our examples. That kind of brings into question whether that's actually an improvement. Having to change the doctests to adhere to the lower accuracies is not a good sign either.

Reply:
In other words, I think this has less to do with how to test, and more with what good default settings are.

Reply:
Ok, I meant to say that if we want to have a discussion around reworking our early-stopping mechanism, we should probably open an issue for it.

----

I'm not particularly uncomfortable with estimators not passing our common tests with the default values. I don't even think we can achieve that.

Take the first failing test, `check_estimators_dtypes`. This one has a dataset with `n_samples=20` and 3 classes. The default for `validation_fraction` is .1 which leads to test_size=2, so `train_test_split` fails with `The test_size = 2 should be greater or equal to the number of classes = 3`

What is the right course of action here? I don't think it's worth changing the default, or having a "clever" 'auto' value just for this test to pass. It's not worth changing the test either. This would be another form of  the VW thing that you mentionned before. The test isn't even remotely related to accuracy/overfitting (so it's not related to early stopping either: there's no harm in disabling it)